staging/src/k8s.io/client-go/tools/cache/controller_test.go

```diff
@@ -893,3 +895,209 @@ func (m *mockTxnStore) Transaction(txns ...Transaction) *TransactionError {
 	m.succeedCount = len(txns)
 	return nil
 }
+
+func TestReplaceEvents(t *testing.T) {
+	_, ctx := ktesting.NewTestContext(t)
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	source := fcache.NewFakeControllerSource()
+	store := NewStore(DeletionHandlingMetaNamespaceKeyFunc)
+	t.Cleanup(func() {
+		source.Shutdown()
+	})
+
+	recorder := &eventRecorder{
+		updateCh: make(chan bool, 1),
```

Comment:
Could delegate constructions to separate function so we don't need to remember to make updateCh?